cli/cmd/policy.go

```diff
@@ -0,0 +1,202 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	policyCmdState = struct {
+		File string
+		Repo bool
+		URL  string
+	}{}
+
+	// policyCmd represents the policy parent command
+	policyCmd = &cobra.Command{
+		Hidden: true,
+		Use:    "policy",
+		Short:  "manage policies",
+		Long: `Manage policies.
+
+NOTE: This feature is not yet available!`,
+	}
+
+	// policyCreateCmd represents the policy create command
+	policyCreateCmd = &cobra.Command{
+		Use:   "create",
+		Short: "create a policy",
+		Long: `Create a policy.
+
+A policy is represented in JSON format.
+The following attributes are minimally required:
+{
+    "policy_id": "lacework-example-1",
+    "title": "My Policy",
+    "enabled": false,
+    "lql_id": "MyQuery",
+    "severity": "high",
+    "description": "My Policy Description",
+    "remediation": "My Policy Remediation"
+}`,
+		Args: cobra.NoArgs,
+		RunE: createPolicy,
+	}
+)
+
+func init() {
+	// add the lql command
+	rootCmd.AddCommand(policyCmd)
+
+	// add sub-commands to the lql command
+	policyCmd.AddCommand(policyCreateCmd)
+
+	// run specific flags
+	setPolicySourceFlags(policyCreateCmd)
+}
+
+func setPolicySourceFlags(cmds ...*cobra.Command) {
+	for _, cmd := range cmds {
+		if cmd == nil {
+			return
+		}
+		action := strings.Split(cmd.Use, " ")[0]
+
+		// file flag to specify a policy from disk
+		cmd.Flags().StringVarP(
+			&policyCmdState.File,
+			"file", "f", "",
+			fmt.Sprintf("path to a policy to %s", action),
+		)
+		/* repo flag to specify a policy from repo
+		cmd.Flags().BoolVarP(
+			&policyCmdState.Repo,
+			"repo", "r", false,
+			fmt.Sprintf("id of a policy to %s via active repo", action),
+		)*/
+		// url flag to specify a policy from url
+		cmd.Flags().StringVarP(
+			&policyCmdState.URL,
+			"url", "u", "",
+			fmt.Sprintf("url to a policy to %s", action),
+		)
+	}
+}
+
+// for commands that take a policy as input
+func inputPolicy(cmd *cobra.Command, args []string) (string, error) {
+	// if running via repo
+	if policyCmdState.Repo {
+		return inputPolicyFromRepo()
+	}
+	// if running via file
+	if policyCmdState.File != "" {
+		return inputPolicyFromFile(policyCmdState.File)
+	}
+	// if running via URL
+	if policyCmdState.URL != "" {
+		return inputPolicyFromURL(policyCmdState.URL)
+	}
+	// if running via editor
+	action := strings.Split(cmd.Use, " ")[0]
+	return inputPolicyFromEditor(action)
+}
+
+func inputPolicyFromRepo() (policy string, err error) {
+	err = errors.New("NotImplementedError")
+	return
+}
+
+func inputPolicyFromFile(filePath string) (policy string, err error) {
+	fileData, err := ioutil.ReadFile(filePath)
+
+	if err != nil {
+		err = errors.Wrap(err, "unable to read file")
+		return
+	}
+
+	policy = string(fileData)
+	return
+}
```

Comment:
Named returns here doesn't help here, it actually makes it more confusing.
```suggestion
func inputPolicyFromFile(filePath string) (string, error) {
	fileData, err := ioutil.ReadFile(filePath)
	if err != nil {
		return "", errors.Wrap(err, "unable to read file")
	}

	return string(fileData), nil
}
```

Reply:
Fixed.